src/components/configMaps/ConfigMap.tsx

```diff
@@ -189,26 +188,29 @@ export function validateKeyValuePair(arr: KeyValue[]): KeyValueValidated {
             valueError = 'value must not be empty'
             isValid = false
         }
-        if (typeof v === 'string' && !PATTERNS.CONFIG_MAP_KEY.test(k)) {
-            keyError = `Key "${k}" must be of ${PATTERNS.CONFIG_MAP_KEY} format`
+        if (typeof v === 'string' && !PATTERNS.CONFIG_MAP_AND_SECRET_KEY.test(k)) {
+            keyError = `Key '${k}' must be of ${PATTERNS.CONFIG_MAP_AND_SECRET_KEY} format`
             isValid = false
         }
         return [...agg, { k, v, keyError, valueError }]
     }, [])
     return { isValid, arr }
 }
 
-export function ConfigMapForm({ id, appId, name = "", external, data = null, type = 'environment', mountPath = "", isUpdate = true, collapse = null, index: listIndex, update: updateForm }) {
+export function ConfigMapForm({ id, appId, name = "", external, data = null, type = 'environment', mountPath = "", isUpdate = true, collapse = null, index: listIndex, update: updateForm, subPath, filePermission }) {
     const [selectedTab, selectTab] = useState(type === 'environment' ? 'Environment Variable' : 'Data Volume')
     const [isExternalValues, toggleExternalValues] = useState(external)
     const [externalValues, setExternalValues] = useState([])
     const [configName, setName] = useState({ value: name, error: "" })
     const [volumeMountPath, setVolumeMountPath] = useState({ value: mountPath, error: "" })
     const [loading, setLoading] = useState(false)
-    const { envId } = useParams()
+    const { envId } = useParams<{ envId }>()
     const [yamlMode, toggleYamlMode] = useState(true)
-    const { yaml, handleYamlChange, error } = useKeyValueYaml(externalValues, setKeyValueArray, PATTERNS.CONFIG_MAP_KEY, `key must be of format ${PATTERNS.CONFIG_MAP_KEY}`)
+    const { yaml, handleYamlChange, error } = useKeyValueYaml(externalValues, setKeyValueArray, PATTERNS.CONFIG_MAP_AND_SECRET_KEY, `key must be of format ${PATTERNS.CONFIG_MAP_AND_SECRET_KEY}`)
```

Comment:
shouldnt we use text in place of showing pattern

Reply:
done

Reply:
done